test/e2e/basic_test.go

```diff
@@ -365,13 +365,41 @@ func testReconcileGatewayRoutesOnStartup(t *testing.T, data *TestData, isIPv6 bo
 		t.Fatalf("Failed to detect gateway interface name from ConfigMap: %v", err)
 	}
 
-	expectedRtNumMin, expectedRtNumMax := clusterInfo.numNodes-1, clusterInfo.numNodes-1
+	nodeNums := clusterInfo.numNodes
+	var expectedRtNum int
 	switch encapMode {
+	case config.TrafficEncapModeEncap:
+		expectedRtNum = nodeNums - 1
+		t.Logf("Encap mode: expected %d gateway routes (one per peer Node)", expectedRtNum)
+
 	case config.TrafficEncapModeNoEncap:
-		expectedRtNumMin, expectedRtNumMax = 0, 0
+		expectedRtNum = 0
+		t.Log("NoEncap mode: expected 0 gateway route")
 
 	case config.TrafficEncapModeHybrid:
-		expectedRtNumMin = 1
+		currentNodeIPStr := nodeIPv4(0)
+		if isIPv6 {
+			currentNodeIPStr = nodeIPv6(0)
+		}
+
+		podName := getAntreaPodName(t, data, nodeName)
+		cmd := []string{"sh", "-c", fmt.Sprintf("ip -o addr show | grep %s | awk '{print $4}' | tr -d ' \\n'", currentNodeIPStr)}
+		stdout, _, err := data.RunCommandFromPod(antreaNamespace, podName, agentContainerName, cmd)
+		require.NoError(t, err)
+		_, currentNodeIP, err := net.ParseCIDR(stdout)
```

Comment:
using `currentNodeIP` as the variable name is misleading here IMO since it is storing a CIDR / prefix

Reply:
The stored NodeIPs doesn't have a suffix. They are from the Node info like this:

```
status:
  addresses:
  - address: 172.18.0.4
    type: InternalIP
  - address: kind-control-plane
    type: Hostname
```

Reply:
My point was that you were parsing the output of `ip addr` (which includes the prefix length) using `net.ParseCIDR`, and storing the CIDR in a variable called `currentNodeIP`. But that code has been updated now, so the issue no longer exists.